server/emails/templates/BaseEmail.tsx

```diff
@@ -148,6 +162,31 @@ export default abstract class BaseEmail<
     }
   }
 
+  private from(props: S & T) {
+    invariant(
+      env.SMTP_FROM_EMAIL,
+      "SMTP_FROM_EMAIL is required to send emails"
+    );
+
+    const parsedFrom = addressparser(env.SMTP_FROM_EMAIL)[0];
+
+    if (this.category === EmailMessageCategory.Authentication) {
+      const domain = parsedFrom.address.split("@")[1];
+      return {
+        name: parsedFrom.name,
```

Comment:
Would it better to retain the behaviour of overriding `fromName` here?

Reply:
I guess so, although I don't think that combination will ever be needed it seems safer

Reply:
Ok 👍